tessellation/src/path_stroke.rs

```diff
@@ -337,81 +337,189 @@ impl<'l, Output: 'l + GeometryBuilder<Vertex>> StrokeBuilder<'l, Output> {
-            self.output.add_triangle(self.previous_a_id, a_id, b_id);
+            self.output.add_triangle(self.previous_left_id, self.previous_right_id, start_left_id);
+            self.output.add_triangle(self.previous_right_id, start_left_id, start_right_id);
         }
 
         self.previous = self.current;
-        self.prev_normal = na;
-        self.previous_a_id = a_id;
-        self.previous_b_id = c_id;
+        self.prev_normal = normal;
+        self.previous_left_id = end_left_id;
+        self.previous_right_id = end_right_id;
         self.current = to;
 
         if self.nth == 1 {
             self.second = self.previous;
-            self.second_a_id = a_id;
-            self.second_b_id = c_id;
-        }
-
-        if maybe_nc.is_some() {
-            let current = self.current;
-            self.tessellate_angle(current, na, a_id, nb, b_id, nc, c_id);
+            self.second_left_id = start_left_id;
+            self.second_right_id = start_right_id;
         }
 
         self.nth += 1;
     }
 
-    fn tessellate_angle(
-        &mut self,
-        _position: Point,
-        _na: Vec2,
-        a_id: VertexId,
-        _nb: Vec2,
-        b_id: VertexId,
-        _nc: Vec2,
-        c_id: VertexId,
-    ) {
-        // TODO: Properly support all types of angles.
-        self.output.add_triangle(b_id, a_id, c_id);
+    fn tesselate_join(&mut self, to: Point, normal: Vec2) -> (VertexId, VertexId, VertexId, VertexId) {
+        // Calculate which side is at the "front" of the join (aka. the pointy side)
+        let a_line = self.current - self.previous;
+        let b_line = to - self.current;
+        let join_angle = a_line.y.atan2(a_line.x) - b_line.y.atan2(b_line.x);
+        let front_side = if join_angle > 0.0 {
+            Side::Left
+        } else {
+            Side::Right
+        };
+
+        // If the "front" is on the right, invert the normal
+        let normal = match front_side {
+            Side::Left => normal,
+            Side::Right => -normal,
+        };
+
+        // Add a vertex at the back of the join
+        let back_vertex = self.output.add_vertex(
+            Vertex {
+                position: self.current,
+                normal: -normal,
+                advancement: self.length,
+                side: front_side.opposite(),
+            }
+        );
+
+        let (start_vertex, end_vertex) = match self.options.line_join {
+            LineJoin::Miter => {
+                let v = self.output.add_vertex(
+                    Vertex {
+                        position: self.current,
+                        normal: normal,
+                        advancement: self.length,
+                        side: front_side,
+                    }
+                );
+
+                (v, v)
+            }
+
+            LineJoin::Round => {
+                let max_radius_segment_angle = compute_max_radius_segment_angle(0.1 /* line width / 2 */, 0.001/*self.options.tolerance as f64*/);
```

Comment:
For the `line_width` stuff I'm thinking of going this way:
 - Have a `line_width: f32` member in `StrokeOptions` with default value `1.0` that we use for for the flattening.
 - Have an `apply_line_width: bool` member in `StrokeOptions`, with default value `true`. If true we apply the line width to positions, and if false we leave the position on the center of the line. Either way the normal stays the same.

I think that these two modes should give the most convenient thing to work with for both the simple use case and the one where we apply the line width in the shader. What do you think?

No need to implement `apply_line_width` in this PR, we can do this as a followup.